src/plugins/discover/public/application/main/components/top_nav/open_alerts_popover.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useState, useMemo } from 'react';
+import ReactDOM from 'react-dom';
+import { I18nStart } from 'kibana/public';
+import { EuiWrappingPopover, EuiLink, EuiContextMenu } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ISearchSource } from '../../../../../../data/common';
+import { KibanaContextProvider, useKibana } from '../../../../../../kibana_react/public';
+import { DiscoverServices } from '../../../../build_services';
+import { updateSearchSource } from '../../utils/update_search_source';
+
+const container = document.createElement('div');
+let isOpen = false;
```

Comment:
Why is this not managed via component state?

Reply:
is is used in a function that's rendering by `ReactDOM.render` so we can't use component state here, since it's no React component. we need it this way because the popover should be closed when there's a second click and it's open. I'd done like our `Options` Popover. It's not very pretty, but it works.